src/REL/d/a/obj/d_a_obj_soil.cpp

```diff
@@ -1,17 +1,621 @@
+    }
+    if (dScGame_c::currentSpawnInfo.stageName == "F200") {
+        static const mVec3_c posOnSkyloft(-7301, 3768, -9309);
+        if (PSVECSquareDistance(mPosition, posOnSkyloft) < 360000) {
+            mHideWhenCameraNear = true;
+        }
+    }
+    mBoundingBox.Set(mVec3_c(-100, -10, -100), mVec3_c(100, 50, 100));
+    if (mAlwaysLoaded) {
+        s32 roomId = mRoomID;
+        if (addActorToRoom(-1) != 0) {
+            mRoomID = roomId;
+            changeLoadedEntitiesWithSet();
+        }
+    }
+    return SUCCEEDED;
+}
+int dAcOsoil_c::actorPostCreate() {
+    if (mSubtype == SUBTYPE_TBOX) {
+        mStateMgr.changeState(StateID_Soil);
+    } else {
+        if (SceneflagManager::sInstance->checkBoolFlag(mRoomID, mSceneflag) || (mParams >> 0x18 & 1) == 0) {
+            mDowsingStateRelated = 0;
+            if (mSubtype == SUBTYPE_DROP || mSubtype == SUBTYPE_MOGMA) {
+                mInteractionDisabled = true;
+            }
+            mStateMgr.changeState(StateID_Hole);
+        } else {
+            mStateMgr.changeState(StateID_Soil);
+        }
+    }
+    if (mDowsingStateRelated == 2) {
+        mDowsingTarget.initialize(DowsingTarget::SLOT_QUEST, 0, nullptr, 5000);
+        if (mItemSubtype == ITEMTYPE_KEY_PIECE && !DowsingTarget::hasKeyPieceDowsing()) {
+            mDowsingStateRelated = 1;
+            unsetActorProperty(AC_PROP_0x1);
+        } else {
+            mDowsingTarget.doRegister();
+        }
+    }
+    if (mForInsects) {
+        dTgInsect_c* tgInsect = nullptr;
+        do {
+            tgInsect = static_cast<dTgInsect_c*>(fManager_c::searchBaseByProfName(fProfile::TAG_INSECT, tgInsect));
+            if (tgInsect != nullptr) {
+                f32 dist = tgInsect->mPosition.squareDistanceToXZ(mPosition);
+                if (dist < 10000) {
+                    mInsectLink.link(tgInsect);
+                    break;
+                }
+            }
+        } while(tgInsect != nullptr);
+    }
+    if (mSceneflag < 0xFF && !SceneflagManager::sInstance->checkBoolFlag(mRoomID, mSceneflag) && dScGame_c::currentSpawnInfo.stageName == "F401") {
+        mColors[MODEL_DUG_UP].a = 0xFF;
+        mModelIndex = MODEL_DUG_UP;
+        mModels[MODEL_DUG_UP].setTevColorAll(GX_TEVREG1, mColors[MODEL_DUG_UP], false);
+        mLightingInfo.setTev1Color(mColors[MODEL_DUG_UP]);
+        unsetActorProperty(AC_PROP_0x1);
+        mHideCover = true;
+    }
+    if (mItemSubtype == ITEMTYPE_KEY_PIECE || mItemSubtype == ITEMTYPE_DIRECT_GIVE) {
+        unsetActorProperty(AC_PROP_0x1);
+    }
+    // this feels fake
+    f32 five;
+    f32 posZ;
+    f32 newY;
+    f32 posX;
+    f32 posY;
+    posZ = mPosition.z;
+    posX = mPosition.x;
+    posY = mPosition.y;
+    five = 5;
+    newY = posY + five;
+    mPositionCopy2.x = posX;
+    mPositionCopy2.z = posZ;
+    mPositionCopy2.y = newY;
+    mPositionCopy3.x = posX;
+    mPositionCopy3.y = newY;
+    mPositionCopy3.z = posZ;
```

Comment:
generally dealing with mPositionCopy2 and mPositionCopy3 have been odd. CopyFrom/CopyTo inlines sometimes fixes it. 

sometimes it will consolidate the mPosition.y + 5 into one addtion even if its written twice